packages/embark/src/lib/modules/blockchain-client/index.js

```diff
@@ -14,9 +15,13 @@ class BlockchainClient {
 
     // TODO: unclear currently if this belongs here so it's a bit hardcoded for now
     this.events.setCommandHandler("blockchain:client:provider", (clientName, cb) => {
-      const Web3 = require('web3');
-      var web3 = new Web3("ws://localhost:8556");
-      cb(null, web3.currentProvider);
+      this.events.request("blockchain:client:endpoint", (err, endpoint) => {
```

Comment:
Just feels odd that this requests from a completely different module, even though it shares the same name.